File: ts/import-content-model/import_model_create_taxonomy.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: 'KONTENT_AI_ENVIRONMENT_ID',
  apiKey: 'KONTENT_AI_MANAGEMENT_API_KEY'
});

const response = client.addTaxonomy()
  .withData(
    {
      name: 'Blogpost topic',
      codename: 'blog_topic',
      terms: [
        {
          name: 'Sport',
          codename: 'sport',
          terms: [
            {
              name: 'Soccer',
              codename: 'soccer'
            },
            {
              name: 'Ice hockey',
              codename: 'hockey'
            },
            {
              name: 'Rugby',
              codename: 'rugby'
            }
          ]
        },
        {
          name: 'Technology stack',
          codename: 'tech',
          terms: [
            {
              name: 'JavaScript',
              codename: 'js'
            },
            {
              name: 'C#',
              codename: 'c'
            },
            {
              name: 'MVC',
              codename: 'mvc'
            }
          ]
        }
      ]
    }
  )
  .toPromise();

File: net/import-content-model/import_model_create_taxonomy.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var response = await client.CreateTaxonomyGroupAsync(new TaxonomyGroupCreateModel
{
    Name = "Blogpost topic",
    Codename = "blog_topic",
    Terms = new TaxonomyTermCreateModel[]
    {
        new TaxonomyTermCreateModel
        {
            Name = "Sport",
            Codename = "sport",
            Terms = new TaxonomyTermCreateModel[]
            {
                new TaxonomyTermCreateModel
                {
                    Name = "Soccer",
                    ExternalId = "soccer"
                },
                new TaxonomyTermCreateModel
                {
                    Name = "Ice hockey",
                    ExternalId = "hockey"
                },
                new TaxonomyTermCreateModel
                {
                    Name = "Rugby",
                    ExternalId = "rugby"
                },
            }
        },
        new TaxonomyTermCreateModel
        {
            Name = "Technology stack",
            Codename = "tech",
            Terms = new TaxonomyTermCreateModel[]
            {
                new TaxonomyTermCreateModel
                {
                    Name = "Javascript",
                    ExternalId = "js"
                },
                new TaxonomyTermCreateModel
                {
                    Name = "C#",
                    ExternalId = "c"
                },
                new TaxonomyTermCreateModel
                {
                    Name = "MVC",
                    ExternalId = "mvc"
                },
            }
        },
    }
});
